lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp

```diff
@@ -399,7 +399,8 @@ struct MMAV3UseRegOperand
         dotOp.getContext(), /*opIdx=*/0, srcEnc, /*kWidth=*/kWidth);
     auto newTy = RankedTensorType::get(srcTy.getShape(), srcTy.getElementType(),
                                        dotOperandEnc);
-    if (!matchMmaV3AndDotOperandLayout(srcTy, newTy))
+    if (!cvtReordersRegisters(srcTy, newTy) &&
+        !matchMmaV3AndDotOperandLayout(srcTy, newTy))
```

Comment:
`matchMmaV3AndDotOperandLayout` is used here for the special shuffle optimization

Reply:
nit. Might be good to be forward-looking (https://github.com/triton-lang/triton/pull/5419) and fail only if we need to use shared memory